chore: Converted pt-br/web/api/animationevent/animationevent to MD (#7236)
mdn/translated-content: files/pt-br/web/api/animationevent/animationevent/index.md

> ---
> title: AnimationEvent()
> slug: Web/API/AnimationEvent/AnimationEvent
> tags:
>   - AnimationEvent
>   - Apps
>   - CSSOM
>   - Experimental
>   - Referencia
> translation_of: Web/API/AnimationEvent/AnimationEvent
> ---

Reviewer: ```suggestion
---
title: AnimationEvent()
slug: Web/API/AnimationEvent/AnimationEvent
page-type: web-api-constructor
tags:
  - API
  - AnimationEvent
  - CSSOM
  - Constructor
  - Reference
  - Web Animations
translation_of: Web/API/AnimationEvent/AnimationEvent
browser-compat: api.AnimationEvent.AnimationEvent
---
```

> {{APIRef("Web Animations API")}}{{SeeCompatTable}}
> 
> `O construtor AnimationEvent()` retorna o recente criado {{domxref("AnimationEvent")}}, representando um evento em relação a animação.

Reviewer: ```suggestion
{{APIRef("Web Animations API")}}

O construtor **`AnimationEvent()`** retorna o objeto  {{domxref("AnimationEvent")}} recém criado, representando um evento em relação a animação.
```

> ## Síntaxe

Reviewer: ```suggestion
## Sintaxe
```

>     animationEvent = new AnimationEvent(type, {animationName: aPropertyName,
>                                                elapsedTime  : aFloat,
>                                                pseudoElement: aPseudoElementName});

Reviewer: ```suggestion
```js
new AnimationEvent(type)
new AnimationEvent(type, options)
```
```

### Argumentos

> *O construtor `AnimationEvent()`* *também herda argumentos do {{domxref("Event.Event", "Event()")}}.*
> 
> - `type`
>   - : Um {{domxref("DOMString")}} representando o nome do tipo de `AnimationEvent`. É caso sensitivo e pode ser: `'animationstart'`, `'animationend'`, ou `'animationiteration'`.

Reviewer: ```suggestion

- `type`
  - : Uma string representando o nome do tipo de `AnimationEvent`. Diferencia maiúsculas e minúsculas e os browsers a definem como `'animationstart'`, `'animationend'`, ou `'animationiteration'`.
```

> - `animationName` {{optional_inline}}
>   - : Um{{domxref("DOMString")}} contendo o valor do {{cssxref("animation-name")}} propriedade CSS associada com a transição. É por padrão `""`.
> - `elapsedTime` {{optional_inline}}
>   - : Um ponto flutuante dando a quantidade de tempo que a animação esteve rodando, em segundos, quando o evento termina, excluindo qualquer tempo que ela passou pausada. Para um evento `"animationstart"`, `elapsedTime` é `0.0` a não ser que haja um valor negativo para [`animation-delay`](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-delay "The animation-delay CSS property specifies when the animation should start. This lets the animation sequence begin some time after it's applied to an element."), nesse caso o evento terminará com `elapsedTime contendo`  `(-1 * `_delay_`)`. É por padrão `0.0`.
> - `pseudoElement` {{optional_inline}}
>   - : É um `{{domxref("DOMString")}}`, começando com`"::"`, contendo o nome do [pseudo-element](https://developer.mozilla.org/en-US/docs/Web/CSS/Pseudo-elements '/en-US/docs/Web/CSS/Pseudo-elements') que a animação roda. Se a animação não roda em um pseudo-elementomas em um elemento, então temos um _string_ vazio `"" .`É por padrão `""`.

Reviewer: ```suggestion
  - : Um {{domxref("DOMString")}} representando o nome do tipo de `AnimationEvent`. Diferencia maiúsculas e minúsculas e pode ser: `'animationstart'`, `'animationend'`, ou `'animationiteration'`.

- `options` {{opcional_inline}}
  - : Um objeto que, _além das propriedades definidas em {{domxref("Event/Event", "Event()")}}_, possui as seguintes propriedades:
    - `animationName` {{optional_inline}}
      - : Uma string contendo o valor da propriedade CSS {{cssxref("animation-name")}} associada à transição. O padrão é `""`.
    - `elapsedTime` {{optional_inline}}
      - : Um `float` que fornece a quantidade de tempo que a animação está em execução, em segundos, quando este evento foi acionado, excluindo qualquer momento em que a animação foi pausada.
        Para um evento `animationstart`, `elapsedTime` é `0.0`, a menos que haja um valor negativo para {{cssxref("animation-delay")}},
        nesse caso o evento será disparado com `elapsedTime` contendo `(-1 * delay)`. O padrão é `0.0`.
    - `pseudoElement` {{optional_inline}}
      - : Uma string, começando com `"::"`, contendo o nome do [pseudo-element](/pt-BR/docs/Web/CSS/Pseudo-elements) em que a animação é executada. Se a animação não for executada em um pseudoelemento, mas no próprio elemento, especifique uma string vazia: `""`. O padrão é `""`.
```

> ## Especificações
> 
> | Especificações                                                                     | Estado                        | Comentário          |
> | ---------------------------------------------------------------------------------- | ----------------------------- | ------------------- |
> | {{ SpecName('CSS3 Animations', '#AnimationEvent-interface', 'AnimationEvent()') }} | {{ Spec2('CSS3 Animations')}} | Initial definition. |

Reviewer: ```suggestion
## Especificações

{{Specifications}}
```

## Compatibilidade com navegadores

{{Compat}}

> ## Veja também 
> 
> - [Usando animações CSS](https://developer.mozilla.org/en-US/docs/CSS/Using_CSS_animations).
> - Tópicos relacionados a animação CSS, propriedades e regras: [`animation`](https://developer.mozilla.org/en-US/docs/Web/CSS/animation 'The animation CSS property is a shorthand property for animation-name, animation-duration, animation-timing-function, animation-delay, animation-iteration-count and animation-direction.'), [`animation-delay`](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-delay "The animation-delay CSS property specifies when the animation should start. This lets the animation sequence begin some time after it's applied to an element."), [`animation-direction`](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-direction 'The animation-direction CSS property indicates whether the animation should play in reverse on alternate cycles.'), [`animation-duration`](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-duration 'The animation-duration CSS property specifies the length of time that an animation should take to complete one cycle.'), [`animation-fill-mode`](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-fill-mode 'The animation-fill-mode CSS property specifies how a CSS animation should apply styles to its target before and after it is executing.'), [`animation-iteration-count`](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-iteration-count 'The animation-iteration-count CSS property defines the number of times an animation cycle should be played before stopping.'), [`animation-name`](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-name 'The animation-name CSS property specifies a list of animations that should be applied to the selected element. Each name indicates a @keyframes at-rule that defines the property values for the animation sequence.'), [`animation-play-state`](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-play-state "The animation-play-state CSS property determines whether an animation is running or paused. You can query this property's value to determine whether or not the animation is currently running; in addition, you can set its value to pause and resume playback of an animation."), [`animation-timing-function`](https://developer.mozilla.org/en-US/docs/Web/CSS/animation-timing-function 'The CSS animation-timing-function property specifies how a CSS animation should progress over the duration of each cycle. The possible values are one or several <timing-function>.'), [`@keyframes`](https://developer.mozilla.org/en-US/docs/Web/CSS/@keyframes 'The @keyframes CSS at-rule lets authors control the intermediate steps in a CSS animation sequence by establishing keyframes (or waypoints) along the animation sequence that must be reached by certain points during the animation. This gives you more specific control over the intermediate steps of the animation sequence than you get when letting the browser handle everything automatically.')

Reviewer: ```suggestion
## Veja também

- [Usando animações CSS](/pt-BR/docs/Web/CSS/CSS_Animations/Using_CSS_animations)
- Propriedades e regras de CSS relacionadas à animação: {{cssxref("animation")}},
  {{cssxref("animation-delay")}}, {{cssxref("animation-direction")}},
  {{cssxref("animation-duration")}}, {{cssxref("animation-fill-mode")}},
  {{cssxref("animation-iteration-count")}}, {{cssxref("animation-name")}},
  {{cssxref("animation-play-state")}}, {{cssxref("animation-timing-function")}},
  {{cssxref("@keyframes")}}
```

> - The {{domxref("AnimationEvent")}} interface it belongs to.

Reviewer: ```suggestion
- A interface {{domxref("AnimationEvent")}} a que pertence.
```
